WordPress/src/main/java/org/wordpress/android/ui/sitecreation/NewSiteCreationMainVM.kt

```diff
@@ -0,0 +1,61 @@
+package org.wordpress.android.ui.sitecreation
+
+import android.arch.lifecycle.Transformations
+import android.arch.lifecycle.ViewModel
+import org.wordpress.android.util.wizard.WizardManager
+import org.wordpress.android.util.wizard.WizardNavigationTarget
+import org.wordpress.android.util.wizard.WizardState
+import org.wordpress.android.viewmodel.SingleEventObservable
+import java.util.Arrays
+import javax.inject.Inject
+
+data class SiteCreationState(
+    val segmentId: Long? = null,
+    val verticalId: String? = null,
+    val siteTitle: String? = null,
+    val siteTagLine: String? = null
+) : WizardState
+
+typealias NavigationTarget = WizardNavigationTarget<SiteCreationStep, SiteCreationState>
+
+class NewSiteCreationMainVM @Inject constructor() :
+        ViewModel() {
+    private val wizardManager: WizardManager<SiteCreationStep> = WizardManager(
+            // TODO we'll receive this from a server/Firebase config
+            Arrays.asList(
+                    SiteCreationStep.fromString("site_creation_segments"),
+                    SiteCreationStep.fromString("site_creation_verticals"),
+                    SiteCreationStep.fromString("site_creation_domains")
+            )
+    )
+    private var isStarted = false
+    private var siteCreationState = SiteCreationState()
+
+    val navigationTargetObservable: SingleEventObservable<NavigationTarget> = SingleEventObservable(
```

Comment:
I have played with this once again with a different approach and the more I think about it and play with it, the more I feel like that the `LiveData` is really not right tool to handle single events like navigation, toast etc. We should look into alternatives for it as a hack week project or something.

Reply:
Yes, I'm starting to have the same feeling. We've tried several different approaches and all of them have some down-sides. Unless we come up with an ultimate solution, I think we should stop using LiveData for such cases.

Reply:
I've discussed it with Vojta and we all have the same feeling. We are thinking about creating a custom LifeCycleAware Listener/Observable which would solve this issue once and for all.